dsc/pwsh.profile.resource.ps1

```diff
@@ -0,0 +1,149 @@
+        'CurrentUserAllHosts' { $PROFILE.CurrentUserAllHosts}
+        'CurrentUserCurrentHost' { $PROFILE.CurrentUserCurrentHost}
+    }
+
+    return $path
+}
+
+function ExportOperation {
+    $allUserCurrentHost = PopulatePwshResource -profileType 'AllUsersCurrentHost'
+    $allUsersAllHost = PopulatePwshResource -profileType 'AllUsersAllHosts'
+    $currentUserAllHost = PopulatePwshResource -profileType 'CurrentUserAllHosts'
+    $currentUserCurrentHost = PopulatePwshResource -profileType 'CurrentUserCurrentHost'
+
+    # Cannot use the ToJson() method here as we are adding a note property
+    $allUserCurrentHost | Add-Member -NotePropertyName '_name' -NotePropertyValue 'AllUsersCurrentHost' -PassThru | ConvertTo-Json -Compress -EnumsAsStrings
+    $allUsersAllHost | Add-Member -NotePropertyName '_name' -NotePropertyValue 'AllUsersAllHosts' -PassThru | ConvertTo-Json -Compress -EnumsAsStrings
+    $currentUserAllHost | Add-Member -NotePropertyName '_name' -NotePropertyValue 'CurrentUserAllHosts' -PassThru | ConvertTo-Json -Compress -EnumsAsStrings
+    $currentUserCurrentHost | Add-Member -NotePropertyName '_name' -NotePropertyValue 'CurrentUserCurrentHost' -PassThru | ConvertTo-Json -Compress -EnumsAsStrings
+}
+
+function GetOperation {
+    param (
+        [PwshResource] $InputResource
+    )
+
+    $profilePath = GetProfilePath -profileType $InputResource.profileType.ToString()
+
+    $exists = Test-Path $profilePath
+
+    if ($InputResource._exist -and $exists) {
+        $content = Get-Content -Path $profilePath
+        $InputResource.Content = $content
+    }
+    elseif ($InputResource._exist -and -not $exists) {
+        $InputResource.Content = $null
+        $InputResource._exist = $false
+    }
+    elseif (-not $InputResource._exist -and $exists) {
+        $InputResource.Content = Get-Content -Path $profilePath
+        $InputResource._exist = $true
+    }
+    else {
+        $InputResource.Content = $null
+        $InputResource._exist = $false
+    }
+
+    $InputResource.ToJson()
+}
+
+function SetOperation {
+    param (
+        [PwshResource] $InputResource
+    )
+
+    $profilePath = GetProfilePath -profileType $InputResource.profileType.ToString()
+    $profileExists = Test-Path $profilePath
+
+    if ($InputResource._exist) {
+        if ($InputResource.content) {
+            Set-Content -Path $profilePath -Value $InputResource.content
+        }
+        else {
+            Remove-Item -Path $profilePath -Force
+        }
+    }
+    elseif (-not $InputResource._exist -and $profileExists) {
+        Remove-Item -Path $profilePath -Force
+    }
+    elseif (-not $InputResource._exist -and -not $profileExists) {
+        # Do nothing
+    }
+}
+
+$inputJson = $input | ConvertFrom-Json
+
+if ($inputJson) {
+    $InputResource = [PwshResource]::new()
+    $InputResource.profileType = $inputJson.profileType
+    $InputResource.content = $inputJson.content
+    $InputResource._exist = $inputJson._exist
+}
```

Comment:
I think we need to do some more handling here to determine whether or not the user explicitly defined `content` in the input - if they _did_, we want to control that content. If they didn't, we want to leave the content alone.